hyperSpec/R/show.R

```diff
@@ -1,136 +1,354 @@
-#' @export
-#' @seealso [base::print()]
-setMethod("print", signature = signature(x = "hyperSpec"), function(x, range = FALSE, ...) {
-  validObject(x)
-  cat(as.character(x, range = FALSE, ...), sep = "\n")
-  invisible(x)
-})
+# Unit tests -----------------------------------------------------------------
+.test(show) <- function() {
+  context("show")
 
+  test_that("show() works", {
+    expect_output(show(faux_cell))
+  })
 
-# FIXME: logbook is mentioned
-# - `summary()` displays the logbook in addition.
+  test_that("show() gives correct output", {
+    expect_output(show(faux_cell), "hyperSpec object")
+    expect_output(show(faux_cell), "300 data points / spectrum")
+  })
 
-#' @details
-#' - `summary()` ...
-#' @aliases summary summary,hyperSpec-method
-#' @seealso [base::summary()]
-#' @export
-#' @rdname show
-setMethod("summary",
-  signature = signature(object = "hyperSpec"),
-  function(object, ...) {
-    print(object, ...)
-  }
-)
+  test_that("show() does not give certain output", {
+    # Does not contain line starting with "wavelength:"
+    expect_output(show(faux_cell), "^(?!wavelength: ).*", perl = TRUE)
+    # Does does not contain line starting with "data:"
+    expect_output(show(faux_cell), "^(?!data: )", perl = TRUE)
+  })
+}
+
+
+.test(print) <- function() {
+  context("print")
+
+  test_that("print() works", {
+    expect_output(print(faux_cell))
+  })
+
+  test_that("print() gives correct output", {
+    expect_output(print(faux_cell), "hyperSpec object")
+    expect_output(print(faux_cell), "300 data points / spectrum")
+
+    expect_output(print(faux_cell, range = TRUE, include = "data"), " rng ")
+    expect_output(print(faux_cell, include = "wl"), "^wavelength: ")
+
+  })
+
+  test_that("print() does not give certain output", {
+    # Does not contain line starting with "wavelength:"
+    expect_output(print(faux_cell), "^(?!wavelength: )", perl = TRUE)
+    # Does does not contain line starting with "data:"
+    expect_output(show(faux_cell), "^(?!data: )", perl = TRUE)
+
+    expect_output(print(faux_cell, include = "data"), "^(?!hyperSpec object)", perl = TRUE)
+    expect_output(print(faux_cell, include = "data"), "^(?!wavelength:)", perl = TRUE)
+
+    expect_output(print(faux_cell, include = "wl"), "^(?!hyperSpec object)", perl = TRUE)
+    expect_output(print(faux_cell, include = "wl"), "^(?!data: )", perl = TRUE)
+  })
+}
+
+
+.test(summary) <- function() {
+  context("summary")
+
+  test_that("summary() works", {
+    expect_output(summary(faux_cell))
+  })
+
+  test_that("summary() gives correct output", {
+    expect_output(summary(faux_cell), "hyperSpec object")
+    expect_output(summary(faux_cell), "300 data points / spectrum")
+    expect_output(summary(faux_cell), "wavelength: ")
+    expect_output(summary(faux_cell), "data: ")
+
+    expect_output(summary(faux_cell, range = TRUE, include = "data"), " rng ")
+  })
+
+# }
+#
+#
+#
+# .test(as.character) <- function() {
+  context("as.character")
```

Comment:
I mentioned these lines in https://github.com/r-hyperspec/hySpc.testthat/issues/4